app/models/user.rb

```diff
@@ -611,4 +626,63 @@ def contributed_projects_ids
       select(:project_id).
       uniq.map(&:project_id)
   end
+
+  def contribution_year_period
+    streak_date_format(1.year.ago) + ' - ' + streak_date_format(Date.today)
+  end
+
+  def longest_streak_period
+    if self.longest_streak == 0
+      return 'No contribution yet'
+    end
+
+    if self.longest_streak_start_at == self.longest_streak_end_at
+      return streak_date_format(self.longest_streak_start_at)
+    end
+
+    streak_date_format(self.longest_streak_start_at) + ' - ' + streak_date_format(self.longest_streak_end_at)
+  end
+
+  def current_streak_period
+    if self.longest_streak == 0
+      return 'No contribution yet'
+    end
+
+    if self.last_contributed_at + 1 < Date.today
+      return 'Last contributed at ' + streak_date_format(self.last_contributed_at)
+    end
+
+    if self.current_streak == 1
+      return streak_date_format(self.last_contributed_at)
+    end
+
+    streak_date_format(self.last_contributed_at - self.current_streak + 1) + ' - ' + streak_date_format(self.last_contributed_at)
+  end
+
+  def streak
+    if self.last_contributed_at == Date.today
+      return true
+    end
+
+    if self.last_contributed_at != nil && self.last_contributed_at + 1 == Date.today
+      self.current_streak += 1
+    else
+      self.current_streak = 1
+    end
+
+    if self.current_streak > self.longest_streak
+      self.longest_streak = self.current_streak
```

Comment:
Do we need to save the longest_streak? we can also calculate it since we have the start and the end date right? What do you think about that approach?